### PR TITLE
[DM/FEATURE] Generic block layer

### PR DESCRIPTION
**Pull request by GuEe-GUI**
## 拉取/合并请求描述：(PR description)

[
1. Disk and blk device management.
2. Support partitions probe auto.
3. Support DFS and user mode fops, ioctl.
4. Add a cmd for blk info.
5. port to the block for SDIO.
6. remove SDIO rt_mmcsd_blk_init
7. remove SDIO gpt

Depends on: https://github.com/RT-Thread/rt-thread/pull/9573
Depends on: https://github.com/RT-Thread/rt-thread/pull/9594

- BSP:
qemu-vexpress-a9
qemu-virt64-aarch64
rockchip/rk3568

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)
- [x] 如果是新增bsp, 已经添加ci检查到[.github/workflows/bsp_buildings.yml](workflows/bsp_buildings.yml)  详细请参考链接[BSP自查](https://www.rt-thread.org/document/site/#/rt-thread-version/rt-thread-standard/development-guide/bsp-selfcheck/bsp_selfcheck)

### CODE REVIEW

File: components/drivers/block/blk_dev.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * 2023-02-25     GuEe-GUI     first version
+ */
+
+#ifndef __BLK_DEV_H__
+#define __BLK_DEV_H__
+
+#include <rthw.h>
+#include <rtthread.h>
+#include <drivers/blk.h>
+#include <drivers/misc.h>
```
Comment (Rbb666) [inline components/drivers/block/blk_dev.h:17]: 这个头文件是不是忘记加到 rtdevice.h 里面了？

File: components/drivers/block/blk_dev.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * Date           Author       Notes
+ * 2023-02-25     GuEe-GUI     first version
+ */
+
+#ifndef __BLK_DEV_H__
+#define __BLK_DEV_H__
+
+#include <rthw.h>
+#include <rtthread.h>
+#include <drivers/blk.h>
```
[the remark this answers is not in the document]
Comment (GuEe-GUI) [reply components/drivers/block/blk_dev.h:16]: 这部分存在dfs和device头文件相互依赖的冲突，所以需要单独导入，才能访问结构体成员

File: components/drivers/block/blk_dev.h (hunk 1 above)
Comment (GuEe-GUI) [reply components/drivers/block/blk_dev.h:17]: 原因和上面是一样的

Comment (Rbb666) [reply components/drivers/block/blk_dev.h:17]: 明白了

**Review by Rbb666 — APPROVED** [review]

**Pull request merged**